lib/datetime_picker_formfield.dart

```diff
@@ -197,12 +197,14 @@ class _DateTimeFieldState extends FormFieldState<DateTime> {
     super.initState();
     if (widget.controller == null) {
       _controller = TextEditingController(text: format(widget.initialValue));
+      _controller.addListener(_handleControllerChanged);
     }
     if (widget.focusNode == null) {
       _focusNode = FocusNode();
+      _focusNode.addListener(_handleFocusChanged);
     }
-    _effectiveController.addListener(_handleControllerChanged);
-    _effectiveFocusNode.addListener(_handleFocusChanged);
```

Comment:
Removing this seems like playing with fire. I want this widget to work as similarly to TextFormField as possible so I can blame them when things go wrong 😀.

Reply:
The lines replacing this are exactly what they have.  They don’t use the _effectiveController/FocusNode.  Besides if it’s wrong you can blame me but I guarantee it’s correct.

Reply:
Would you like it better if I put these in the else clause instead like TextFormField does?  It's logically the same thing.